subsys/bluetooth/mesh/crypto_tc.c

```diff
@@ -67,20 +75,46 @@ int bt_mesh_aes_cmac(const uint8_t key[16], struct bt_mesh_sg *sg,
 	return 0;
 }
 
+int bt_mesh_pub_key_gen(void)
+{
+	int rc = uECC_make_key(key.public_key_be, key.private_key_be, &curve_secp256r1);
+
+	if (rc == TC_CRYPTO_FAIL) {
+		key.is_ready = false;
+		BT_ERR("Failed to create public/private pair");
+		return -EIO;
+	}
+
+	key.is_ready = true;
+
+	return 0;
+}
+
+const uint8_t *bt_mesh_pub_key_get(void)
+{
+	return key.is_ready ? key.public_key_be : NULL;
```

Comment:
I tried to add key generation in place. It doesn't works since provisionee starts public key sending after it receives public key and generates shared secret. However, to the time of shared secret generation the existing public key is expected.

Reply:
Why it can't generate public key in `prov_dh_key_gen`?

Reply:
Ok, lets keep it as is for now. Will see if we can improve this later.

Reply:
Actually, it is possible but it will change a little bit approach for provisioner and provisionee. Now they are kind of similar.